src/api/app/models/flag.rb

```diff
@@ -4,6 +4,28 @@ class Flag < ActiveRecord::Base
 
   belongs_to :architecture
 
+  validates :flag, :presence => true
+  validates :position, :presence => true
+  validates_numericality_of :position, :only_integer => true
+
+  before_validation(:on => :create) do
+    if self.project
+      self.position = (self.project.flags.maximum(:position) || 0 ) + 1
```

Comment:
I just moved this to the top of the file. Some nasty abuse to refactor for another day... :-)